src/test/auxiliary/needs_allocator.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// no-prefer-dynamic
+
+#![feature(no_std, core, custom_attribute)]
+#![no_std]
+#![needs_allocator]
```

Comment:
This is gated by the `custom_attribute` lint (and there's a test to ensure that `#![needs_allocator]` is gated), but I'll just switch it over to a full-fledged feature and call it a day